src/Sentry/Protocol/Envelopes/Envelope.cs

```diff
@@ -325,6 +326,35 @@ public static Envelope FromTransaction(Transaction transaction)
         return new Envelope(eventId, header, items);
     }
 
+    /// <summary>
+    /// Creates an envelope that contains one or more <see cref="CodeLocations"/>
+    /// </summary>
+    internal static Envelope FromCodeLocations(CodeLocations codeLocations)
+    {
+        var header = DefaultHeader;
+
+        var items = new List<EnvelopeItem>(1);
+        items.Add(EnvelopeItem.FromCodeLocations(codeLocations));
+
+        return new Envelope(header, items);
+    }
+
+    /// <summary>
+    /// Creates an envelope that contains one or more Metrics
+    /// </summary>
+    internal static Envelope FromMetrics(IEnumerable<Metric> metrics)
+    {
+        var header = DefaultHeader;
+
+        List<EnvelopeItem> items = new();
+        foreach (var metric in metrics)
+        {
+            items.Add(EnvelopeItem.FromMetric(metric));
+        }
```

Comment:
```suggestion
        var items = metrics.Select(EnvelopeItem.FromMetric).ToList();
```
We could give LINQ a go. :D

Reply:
I think in an earlier version of the code I was. Bruno suggested removing linq for performance reasons... it's a pity - linq is easier on the eyes (just harder on the CPU).

Reply:
I'd avoid LINQ as a plague in hot paths

Reply:
but worth measuring when it matters, I just find a loop is pretty simple and often default to that if it's hot path